File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        a = []
        temp1 = list1
        temp2 = list2
        while temp1 is not None and temp2 is not None:
            if temp1.val <= temp2.val:
                a.append(temp1.val)
                temp1 = temp1.next
            # elif temp1.val < temp2.val:
            #     a.append(temp1.val)
            #     temp1 = temp1.next
            else:
                a.append(temp2.val)
                temp2 = temp2.next
        while temp1 is not None:
            a.append(temp1.val)
            temp1 = temp1.next
        while temp2 is not None:
            a.append(temp2.val)
            temp2 = temp2.next
        
        if not a :
            return None

        newNode = ListNode(a[0])
        current = newNode
        for i in range(1,len(a)):
            current.next = ListNode(a[i])
            current = current.next
        return newNode

            
        

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode temp1 = list1;
        ListNode temp2 = list2;
        ListNode temp = new ListNode(0);
        ListNode curr = temp;
        while ( temp1 != null && temp2 != null){
            if (temp1.val<= temp2.val){
                curr.next = temp1;
                temp1 =temp1.next;
            }
            else {
                curr.next = temp2;
                temp2 = temp2.next;
            }
            curr = curr.next;
            
        }
        curr.next =(temp1 !=null) ? temp1:temp2;
        return temp.next;
    }
}